### ISSUE TITLE
Domain Picker: split styles into smaller files

### ISSUE THREAD

**Issue by ciampo** [Automattic/wp-calypso#51187]
## What

As a follow-up task after #51144, we should split the Domain Picker styles into smaller files, closer to each subcomponent

## Why

This will increase modularity and decouple the different parts of the Domain Picker component

## A/C

- [ ] Issue #51144 has been solved
- [ ] The ignored eslint rule about CSS classnames matching a file's path should be enabled again, and the necessary changes to classnames should be applied
- [ ] Each component's styles should belong to a `style.scss` file in the same folder as the component's markup

## Context

N/A
